### ISSUE TITLE
View images from album, how can I use future data

### ISSUE THREAD

**Issue by cherrybiu** [fireslime/photo_view#340]
when I use `PhotoViewGalleryPageOptions`, the imageProvider can accepts the `MemoryImage`, but the data is Future<Uint8List>, it cant be accepted. I have seen several similar questions, and you replied it can be solved by `'PhotoView' `. But it seems can only solve single image, not gallery. Could you tell me how to deal the gallery preview?

**Comment by clragon** [comment]
if you use `PhotoViewGalleryPageOptions.customChild`, you can provide a `child` widget, instead of an image provider.
In this case, you can then provide a PhotoView widget as the child.

**Comment by cherrybiu** [comment]
> u can then provide a PhotoView widget as the child

yeah，this way can did solve the problem. Actually, I used customChild in the beginning,but this raised another problem that the double tap to change the scale cant work.It works only I set the childSize property.  I cant set the childSize, cause the pic needed to fill the screen. Once I set the childSize to MediaQuery.of(content).size, it can't work too. They look contradictoy.
